src/main/kotlin/com/cognifide/gradle/aem/common/pkg/vlt/VltDefinition.kt

```diff
@@ -39,27 +38,41 @@ open class VltDefinition(private val aem: AemExtension) {
     var createdBy: String? = System.getProperty("user.name")
 
     @Internal
-    var filterElements: MutableList<Element> = mutableListOf()
+    var filterElements = mutableListOf<FilterElement>()
 
     @get:Internal
-    val filterRoots: List<String>
-        get() = filterElements.map { it.attr("root") }
+    val filterEffectives: Collection<FilterElement>
+        get() = filterElements.asSequence()
+                .filter { custom ->
```

Comment:
Please name this filter lambda because it is hard to follow - too long for lambda without any name.

Reply:
methods extracted, done